tests/unit/fem/ptest_tdofs.cpp

```diff
@@ -0,0 +1,1012 @@
+
+         auto ldof = decodeDof(idofs[0]);
+         CHECK( ldof >= 0);
+
+         auto tdof = fes.GetGlobalTDofNumber(ldof);
+         CHECK( tdof >= 0);
+
+         face_tdof[f] =tdof;
+
+         // generate some debug output
+         if (m_verbose)
+         {
+            sstr<< std::endl <<"\t"
+                <<"{face:" << f
+                <<",l:" << ldof
+                <<",T_l:" << fes.GetLocalTDofNumber(ldof)
+                <<",T_g:" << tdof
+                <<"}";
+         }
+      }
+      MPRINT(rank, sstr.str());
+
+      // check that all edges have valid tdofs
+      checkDOFs(face_tdof, "face");
+
+      delete fec;
+   }
+
+   /// Generate and test tdofs for the mesh elements
+   void testElementTDofs()
+   {
+      const int dim = m_mesh->Dimension();
+      const int order = 0;
+
+      const int NE = m_mesh->GetNE();
+
+      MPRINT_MASTER(rank,"Finding element tdofs");
+
+      auto* fec = new mfem::L2_FECollection(order, dim);
+      mfem::ParFiniteElementSpace fes(m_mesh, fec);
+      mfem::Array<int> elem_tdof(NE);
+      elem_tdof = UNITIALIZED_VAL;
+
+      mfem::Array<int> idofs;
+
+      std::stringstream sstr;
+      for (int el=0; el< NE; ++el)
+      {
+         fes.GetElementDofs(el, idofs);
+         CHECK( idofs.Size() == 1);
+
+         auto ldof = idofs[0];
+         CHECK( ldof >= 0);
+
+         auto tdof = fes.GetGlobalTDofNumber(ldof);
+         CHECK( tdof >= 0);
+
+         elem_tdof[el] =tdof;
+
+         // generate some debug output
+         if (m_verbose)
+         {
+            sstr<< std::endl <<"\t"
+                <<"{element:" << el
+                <<",l:" << ldof
+                <<",T_l:" << fes.GetLocalTDofNumber(ldof)
+                <<",T_g:" << tdof
+                <<"}";
+         }
+      }
+      MPRINT(rank, sstr.str());
+
+      // check that all elements have valid tdofs
+      checkDOFs(elem_tdof, "element");
+
+      delete fec;
+   }
+
+
+   /// Generate and test tdofs for the mesh boundary elements
+   void testBoundaryElementTDofs()
+   {
+      MPRINT_MASTER(rank,"Finding boundary element tdofs");
+
+      const int dim = m_mesh->Dimension();
+      const int order = 0;
+      const int NBE = m_mesh->GetNBE();
+
+      int NCodimFaces = 0;
+      switch (dim)
+      {
+         case 1: NCodimFaces = m_mesh->GetNV(); break;
+         case 2: NCodimFaces = m_mesh->GetNEdges(); break;
+         case 3: NCodimFaces = m_mesh->GetNFaces(); break;
+         default: FAIL("Dim must be 1,2 or 3."); break;
+      }
+
+      auto* fec = new mfem::RT_FECollection(order, dim);
+      mfem::ParFiniteElementSpace fes(m_mesh, fec);
+      mfem::Array<int> be_tdof(NBE);
+      be_tdof = UNITIALIZED_VAL;
+
+      mfem::Array<int> idofs;
+
+      std::stringstream sstr;
+      for (int be=0; be< NBE; ++be)
+      {
+         // Find tdof of mesh element corresponding to the boundary element
+         auto elem_id = m_mesh->GetBdrElementEdgeIndex(be);
```

Comment:
I couldn't get this to work with ``FiniteElementSpace::GetBdrElementDofs()`` on ``RT`` spaces.
It seemed like it should work, but it was segfaulting. Perhaps there is a missing table?

My workaround was to combine the mapping from boundary elements to their corresponding mesh element index ( ``m_mesh->GetBdrElementEdgeIndex(be)``) with the corresponding call to get the dof from that element (``fes.Get{Vertex,Edge,Face}Dofs()``).